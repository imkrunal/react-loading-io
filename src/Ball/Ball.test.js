import React from 'react'
import serializer from 'jest-emotion'
import renderer from 'react-test-renderer'
import { Ball } from '..'

expect.addSnapshotSerializer(serializer)

it('ball renders correctly', () => {
  const tree = renderer.create(<Ball />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('px converts', () => {
  const tree = renderer.create(<Ball size={10} sizeUnit="em" />).toJSON()
  expect(tree).toMatchSnapshot()
})
