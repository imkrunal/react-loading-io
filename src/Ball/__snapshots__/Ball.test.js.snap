// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ball renders correctly 1`] = `
@keyframes animation-0 {
  0%, 100% {
    -webkit-animation-timing-function: cubic-bezier(0.45,0,0.9,0.55);
    animation-timing-function: cubic-bezier(0.45,0,0.9,0.55);
  }

  0% {
    -webkit-transform: translate(0,0);
    -ms-transform: translate(0,0);
    transform: translate(0,0);
  }

  50% {
    -webkit-transform: translate(0,108px);
    -ms-transform: translate(0,108px);
    transform: translate(0,108px);
    -webkit-animation-timing-function: cubic-bezier(0,0.45,0.55,0.9);
    animation-timing-function: cubic-bezier(0,0.45,0.55,0.9);
  }

  100% {
    -webkit-transform: translate(0,0);
    -ms-transform: translate(0,0);
    transform: translate(0,0);
  }
}

.emotion-1 {
  position: relative;
  width: 200px !important;
  height: 200px !important;
  -webkit-transform: translate( -200px,-200px ) scale(1) translate( 200px,200px );
  -ms-transform: translate( -200px,-200px ) scale(1) translate( 200px,200px );
  transform: translate( -200px,-200px ) scale(1) translate( 200px,200px );
}

.emotion-0 {
  position: absolute;
  -webkit-animation: animation-0 1s linear infinite;
  animation: animation-0 1s linear infinite;
  width: 52px;
  height: 52px;
  top: 20px;
  left: 74px;
  border-radius: 50%;
  background: #f08d43;
}

<div>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-0"
    />
  </div>
</div>
`;

exports[`px converts 1`] = `
@keyframes animation-0 {
  0%, 100% {
    -webkit-animation-timing-function: cubic-bezier(0.45,0,0.9,0.55);
    animation-timing-function: cubic-bezier(0.45,0,0.9,0.55);
  }

  0% {
    -webkit-transform: translate(0,0);
    -ms-transform: translate(0,0);
    transform: translate(0,0);
  }

  50% {
    -webkit-transform: translate(0,108px);
    -ms-transform: translate(0,108px);
    transform: translate(0,108px);
    -webkit-animation-timing-function: cubic-bezier(0,0.45,0.55,0.9);
    animation-timing-function: cubic-bezier(0,0.45,0.55,0.9);
  }

  100% {
    -webkit-transform: translate(0,0);
    -ms-transform: translate(0,0);
    transform: translate(0,0);
  }
}

.emotion-0 {
  position: absolute;
  -webkit-animation: animation-0 1s linear infinite;
  animation: animation-0 1s linear infinite;
  width: 52px;
  height: 52px;
  top: 20px;
  left: 74px;
  border-radius: 50%;
  background: #f08d43;
}

.emotion-1 {
  position: relative;
  width: 10em !important;
  height: 10em !important;
  -webkit-transform: translate( -8em,-8em ) scale(0.8) translate( 8em,8em );
  -ms-transform: translate( -8em,-8em ) scale(0.8) translate( 8em,8em );
  transform: translate( -8em,-8em ) scale(0.8) translate( 8em,8em );
}

<div>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-0"
    />
  </div>
</div>
`;
